content-services/content-sources-frontend
# Refs 1881: Add deeplink and routing to modals

## Summary
This adds deep linking and routing to all modals. 

- Modals all have their own paths that are traversable like any other page.
- If one attempts to navigate to a modal with a bad path (for example: content/add-repositry instead of content/add-repository) or to a modal that requires permissions the user doesn't have; The path should immediately change to the parent path, without showing any modal.
- If one attempts to navigate to a modal path with bad paramaters (for example: content/edit-repository?repoUUIDS=my-car-is-not-a-uuid) the modal will attempt to fetch 3 times before closing and showing an error. 

Additional changes: 
- Removed [ hide-package feature flag](https://issues.redhat.com/browse/HMS-630) 
- Standardized all error notifications
- Standardized hook structure (Hooks folder code clean-up*)
- Rewrote or fixed all UI tests around modals.
- "Need help getting started with Preview features?" now has a saved state so users won't have to keep closing it if not interested.
- Fixed an issue with empty table state on repositories not appearing correctly.
- Fixed a few issues on the popular repositories page around empty table state: 
    - Pagination now hides correctly
    - Clicking clear button now immediately falls back to the "empty" cached page.

## Testing steps

- Use the above explanation to examine all modals for incorrect behaviour.
- As much of the Edit modal specifically was re-written, please ensure that functionality there is as expected (I have reviewed it but would prefer a second pair of eyes).
